components/src/preact/textInput/text-input.tsx

```diff
@@ -55,43 +55,149 @@ const TextInputInner: FunctionComponent<TextInputInnerProps> = ({ lapisField, pl
         return <NoDataDisplay />;
     }
 
-    const onInput = () => {
-        const value = inputRef.current?.value === '' ? undefined : inputRef.current?.value;
-
-        if (isValidValue(value)) {
-            inputRef.current?.dispatchEvent(
-                new CustomEvent('gs-text-input-changed', {
-                    detail: { [lapisField]: value },
-                    bubbles: true,
-                    composed: true,
-                }),
-            );
-        }
-    };
+    return (
+        <TextSelector
+            lapisField={lapisField}
+            initialValue={initialValue}
+            placeholderText={placeholderText}
+            data={data}
+        />
+    );
+};
 
-    const isValidValue = (value: string | undefined) => {
-        if (value === undefined) {
+const TextSelector = ({
+    lapisField,
+    initialValue,
+    placeholderText,
+    data,
+}: TextInputInnerProps & {
+    data: string[];
+}) => {
+    const [items, setItems] = useState(data.filter((item) => filterByInputValue(item, initialValue)));
+
+    const divRef = useRef<HTMLDivElement>(null);
+
+    const shadowRoot = divRef.current?.shadowRoot ?? undefined;
+
+    const environment =
+        shadowRoot !== undefined
+            ? {
+                  addEventListener: window.addEventListener.bind(window),
+                  removeEventListener: window.removeEventListener.bind(window),
+                  document: shadowRoot.ownerDocument,
+                  Node: window.Node,
+              }
+            : undefined;
+
+    function filterByInputValue(item: string, inputValue: string | undefined | null) {
+        if (inputValue === undefined || inputValue === null || inputValue === '') {
             return true;
         }
-        return data.includes(value);
+        return (
+            item?.toLowerCase().includes(inputValue?.toLowerCase() || '') ||
+            item?.toLowerCase().includes(inputValue?.toLowerCase() || '')
+        );
+    }
+
+    const {
+        isOpen,
+        getToggleButtonProps,
+        getMenuProps,
+        getInputProps,
+        highlightedIndex,
+        getItemProps,
+        selectedItem,
+        inputValue,
+        selectItem,
+        setInputValue,
+        closeMenu,
+    } = useCombobox({
+        onInputValueChange({ inputValue }) {
+            setItems(data.filter((item) => filterByInputValue(item, inputValue)));
+        },
+        onSelectedItemChange({ selectedItem }) {
+            if (selectedItem !== null) {
+                divRef.current?.dispatchEvent(new TextChangedEvent({ [lapisField]: selectedItem }));
+            }
+        },
+        items,
+        itemToString(item) {
+            return item || '';
+        },
+        initialSelectedItem: initialValue,
+        environment,
+    });
+
+    const onInputBlur = () => {
+        if (inputValue === '') {
+            divRef.current?.dispatchEvent(new TextChangedEvent({ [lapisField]: null }));
```

Comment:
then this needs to be changed to 
```suggestion
            divRef.current?.dispatchEvent(new TextChangedEvent({ [lapisField]: undefined }));
```